app/src/main/java/com/sdp13epfl2021/projmag/activities/TagsSelectorActivity.kt

```diff
@@ -0,0 +1,78 @@
+package com.sdp13epfl2021.projmag.activities
+
+import android.content.Intent
+import android.graphics.Color
+import android.graphics.Color.blue
+import android.os.Bundle
+import android.view.View
+import android.widget.Button
+import androidx.appcompat.app.AppCompatActivity
+import androidx.recyclerview.widget.RecyclerView
+import com.sdp13epfl2021.projmag.R
+import com.sdp13epfl2021.projmag.activities.listenerClass.RecyclerItemClickListenr
+import com.sdp13epfl2021.projmag.adapter.TagAdapter
+import com.sdp13epfl2021.projmag.model.Tag
+import com.sdp13epfl2021.projmag.model.TagsBaseManager
+import java.io.Serializable
+
+
+class TagsSelectorActivity : AppCompatActivity() {
+    private val selectedTags : MutableSet<Tag> = mutableSetOf()
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_tags_selector)
+
+        val manager = TagsBaseManager()
+        val tagsDataset = manager.getAllTags().toList()
+        val tagRecyclerView = findViewById<RecyclerView>(R.id.recycler_tag_view)
+        val saveButton = findViewById<Button>(R.id.DoneTagButton)
+        tagRecyclerView.adapter = TagAdapter(this, tagsDataset)
+        tagRecyclerView.setHasFixedSize(true)
+
+
+        tagRecyclerView.addOnItemTouchListener(
+            RecyclerItemClickListenr(
+                this,
+                tagRecyclerView,
+                object : RecyclerItemClickListenr.OnItemClickListener {
+
+                    override fun onItemClick(view: View, position: Int) {
+                        if(selectedTags.contains(tagsDataset[position])) {
+                            tagRecyclerView.findViewHolderForLayoutPosition(position)?.itemView?.alpha = 0.5f
+                            selectedTags.remove(tagsDataset[position])
+                            //need to add color change somehow
+                        }
+                        else{
+                            tagRecyclerView.findViewHolderForLayoutPosition(position)?.itemView?.alpha = 1f
```

Comment:
This will change the tag color as you said you were looking.
```suggestion
                        val holder = tagRecyclerView.findViewHolderForLayoutPosition(position) as TagAdapter.TagViewHolder
                        if(selectedTags.contains(tagsDataset[position])) {
                            holder.itemView.alpha = 0.5f
                            holder.textView.setTextColor(Color.RED)
                            selectedTags.remove(tagsDataset[position])
                        }
                        else{
                            holder.itemView.alpha = 1f
                            holder.textView.setTextColor(Color.GREEN)
```